Tests/AutoTests/FunctionalTest/functional-weeklyfiledownload.spec.ts

```diff
@@ -25,6 +28,19 @@ test.describe("Goto maritime-safety-information Notice To Mariner Page to Check
     expect(newPageUrl).toContain(`NoticesToMariners/DownloadFile?fileName=${fileName}`);
   })
 
+  test('Should Goto Notices to Mariner Page for Weekly Download with Distributor Role', async ({ page, context }) => {
+    await login.goToSignIn();
+    await login.loginWithDistributorDetails(app.DistributorTest_UserName, app.DistributorTest_Password);
+    await noticeFileDownload.goToNoticeToMariner();
+    await noticeFileDownload.checkWeeklyFileSectionName();
+    await noticeFileDownload.checkWeeklyFileSortingWithDistributorRole();
+    const name = await noticeFileDownload.checkFileDownload();
+    expect((await name).length).toBeGreaterThan(0);
+    var fileName = name[0];
+    const newPageUrl = await (await page.$("[id^='download'] > a")).getAttribute('href');
```

Comment:
There should be no selectors etc. in the test.  All knowledge of page structure should go in the page object class